File: 3743-reschedule-meetings-for-maximum-free-time-i/3743-reschedule-meetings-for-maximum-free-time-i.py
class Solution:
    def maxFreeTime(self, eventTime: int, k: int, start: List[int], end: List[int]) -> int:
        g=[]
        g.append(start[0])
        for i in range(1,len(start)):
            g.append(start[i]-end[i-1])
        g.append(eventTime-end[len(end)-1])
        s=result=sum(g[:k+1])
        for i in range(k+1,len(g)):
            s+=g[i]-g[i-k-1]
            result=max(result,s)
        return result

File: 3743-reschedule-meetings-for-maximum-free-time-i/3743-reschedule-meetings-for-maximum-free-time-i.java
class Solution {
    public int maxFreeTime(int eventTime, int k, int[] startTime, int[] endTime) {
        List<Integer> list= new ArrayList<>();
        list.add(startTime[0]);
        int n=startTime.length;
        for(int i=1;i<n;i++){
            list.add(startTime[i]-endTime[i-1]);
        }
        list.add(eventTime-endTime[n-1]);
        int result=0,cur=0;
        for(int i=0;i<list.size();i++){
            cur+=list.get(i);
            if(i>=k+1) cur-=list.get(i-k-1);
            result=Math.max(result,cur);
        }
        return result;
    }
}